### DISCUSSION TITLE
Should KaTeX work within SVG text elements?

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [KaTeX/KaTeX#3287] [Q&A] [answered]
I'm working on producing graphs and am building everything in SVG; however, when I try to include KaTeX in the SVG `text` element, it doesn't show up: is this expected? or does it indicate something is wrong with my current set up? 

Just looking for confirmation as to the expected behaviour so I know where to get started. Thanks in advance for your time and help!

**Comment by edemaine** [comment, marked as answer]
No, it shouldn't work. KaTeX needs an HTML environment to render. You can escape from SVG to HTML (if you're using an HTML viewer) via `<foreignObject>`, but `<text>` remains in SVG.

FWIW, MathJax supports pure SVG output. For KaTeX, it's feature request issue #375.

**Reply by JacobBeaudry** [reply]
Thanks for the very quick & clear reply! Greatly appreciate it! If I find any useful solutions, I'll be sure to share, here.
